src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/category/CategoriesFieldset.java

```diff
@@ -0,0 +1,243 @@
+  }
+
+  public CategoryPill.Data findCategoryOrElseThrow(final String categoryName)
+    throws NotFoundException {
+    return findCategoryWith(categoryName)
+      .orElseThrow(() -> new NotFoundException(CATEGORY_NOT_FOUND))
+      .toData();
+  }
+
+  /**
+   *
+   * @param categoryName to match
+   * @return first WebElement category that matches the name
+   */
+  private Optional<WebElement> getCategory(List<WebElement> categoryList, final String categoryName)
+    throws NotFoundException {
+    List<WebElement> foundCategories = categoryList.stream()
+      .filter(element -> element
+        .getAttribute("innerText")
+        .trim()
+        .equalsIgnoreCase(categoryName)
+      )
+      .collect(Collectors.toList());
+
+      // because of differences in mobile and desktop views
+      if (foundCategories.isEmpty()) {
+        return Optional.empty();
+      } else if (foundCategories.size() == 1) {
+        return Optional.of(foundCategories.get(0));
+      } else {
+        return foundCategories
+          .stream()
+          .filter(WebElement::isDisplayed)
+          .findFirst();
+      }
+  }
+
+  private int getCategoryPosition(List<WebElement> categoryList, final String categoryName)
+    throws NotFoundException {
+    WebElement category = getCategory(categoryList, categoryName)
+      .orElseThrow(()-> new NotFoundException(CATEGORY_NOT_FOUND));
+    return categoryList.indexOf(category);
+  }
+
+  private int getCategoryPosition(final String categoryName) {
+    return getCategoryPosition(this.categories, categoryName);
+  }
+
+  private Optional<WebElement> getCategoryWith(String categoryName) {
+    return getCategory(this.categories, categoryName);
+  }
+
+  /**
+   *
+   * @param categoryName to look for
+   * @return this category on list of editable categories
+   * Categories on editable categories list don't have any attribute based on their names,
+   * and thus a lookup in non-editable categories list must be made beforehand
+   */
+  private WebElement getEditableCategoryWith(String categoryName) {
+    return editableCategories.get(getCategoryPosition(categories, categoryName));
+  }
+
+  public WebElement getEditableCategoriesDesktop(String categoryName) {
+    int position = getCategoryPosition(categoryName);
+    this.clickEdit();
+    return editableCategories.get(position);
+  }
+
+  public CategoriesFieldset clickCategoryWith(final String categoryName) {
+    PageObjectLogging.log("Clicking on category", categoryName, true);
+    getCategoryWith(categoryName)
+      .orElseThrow(() -> new NotFoundException(CATEGORY_NOT_FOUND))
+      .click();
+    return this;
+  }
+
+  public CategoriesFieldset clickEdit() {
+    editLink.click();
+    return this;
+  }
+
+  /**
+   * For mobile "All" category is not visible, that is why it's always false - can not be edited.
+   * TODO: please remember to change this line when SOC-3793 is done:
+   * TODO: "isCategoryEditable(editableCategoryAll)"
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1135)